src/pages/index.js

```diff
@@ -170,16 +242,187 @@ profileCard.setEventListeners();
+
 const newCard = new PopupWithForm({
   popupSelector: "#add-card-modal",
   handleFormSubmit: (inputValues) => {
-    console.log(inputValues);
-    const card = renderCard(inputValues);
-    sectionRenderer.addItem(card);
-    // const name = cardTitleInput.value;
-    // const link = cardUrlInput.value;
-    newCard.close();
+    const name = inputValues.name;
+    const link = inputValues.link;
+    api
+      .addNewCards({ name, link })
+      .then((res) => {
+        console.log(res);
+        console.log(inputValues);
+        // "res" in the renderCard parameter takes all the data from the .then("res") and pushes all of the necessary info that the server has to the cards
+        const card = renderCard(res);
+        sectionRenderer.addItem(card);
+        newCard.close();
+      })
+      .catch((err) => {
+        console.error("an error has occurred", err);
+      })
+      .finally(() => {
+        console.log("done");
+      });
   },
 });
 
 newCard.setEventListeners();
+
+/* -------------------------------------------------------------------------- */
+/*                             confirmation popup                             */
+/* -------------------------------------------------------------------------- */
+
+const popupConfirm = new PopupWithConfirmation({
+  popupSelector: "#confirm-delete-modal",
+});
+
+function handleDeleteButton(card) {
+  popupConfirm.setSubmitAction(() => {
+    api
+      .confirmDeleteButton(card._id)
+      .then(() => {
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.error("an error has occurred", err);
+      })
+      .finally(() => {
+        console.log("done");
+      });
+  });
+  popupConfirm.open();
+}
+
+popupConfirm.setEventListeners();
+
+/* -------------------------------------------------------------------------- */
+/*                   api call for the section card renderer                   */
+/* -------------------------------------------------------------------------- */
+
+let sectionRenderer;
+
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    sectionRenderer = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          const card = renderCard(cardData);
+          sectionRenderer.addItem(card);
+        },
+      },
+      cardListSelector
+    );
+    sectionRenderer.renderItems();
+    const cards = initialCards;
+  })
+  //need to figure out a way to take the ids and plug them into the url. Do i use an event listener of every card and everytime i click a specific one i get the id for it i do i just listen for every time?
+  .catch((err) => {
+    console.error("An error was found", err);
+  });
+
+/* -------------------------------------------------------------------------- */
+/*                                fetch request                               */
+/* -------------------------------------------------------------------------- */
+api
+  .userProfileInfo()
```

Comment:
You need to combine the requests of the user's info and the cards together in `Promise.all` because it might happen so that the `_id` has not been received yet but the cards have and your cards will be rendered not correctly. 
More info is here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

```js

Promise.all([getUserInfo(), getCards()])
// destructure the response
  .then(([userData, cards]) => {
      // set all the data
  })
  .catch(err => {
    // catch possible errors
  });

```